results_wip.py

```diff
@@ -0,0 +1,597 @@
+                'No code coverage increasement, indicating the fuzz target'
+                ' ineffectively invokes the function under test.',
+            self.NON_SEC_CRASH_NULL_DEREF:
+                'Accessing a null pointer, indicating improper parameter '
+                'initialization or incorrect function usages in the fuzz target.',
+            self.NON_SEC_CRASH_SIGNAL:
+                'Abort with signal, indicating the fuzz target has violated some '
+                'assertion in the project, likely due to improper parameter '
+                'initialization or incorrect function usages.',
+            self.NON_SEC_CRASH_EXIT:
+                'Fuzz target exited in a controlled manner without showing any '
+                'sign of memory corruption, likely due to the fuzz target is not '
+                'well designed to effectively find memory corruption '
+                'vulnerability in the function-under-test.',
+            self.GEN_OVERWRITE_CONST:
+                'Fuzz target modified a const data. To fix this, ensure that all '
+                'input data passed to the fuzz target is treated as read-only '
+                'and not modified. Copy the input data to a separate buffer if '
+                'any modifications are necessary.',
+        }
+
+        return mapping.get(self, self.NORM_NOT_APPLICABLE)
+
+
+@dataclass
+class BuildInfo:
+    """Information about a fuzz target build."""
+    compiles: bool = False
+    log_path: str = ""
+    errors: List[str] = field(default_factory=list)
+    binary_exists: bool = False
+    is_function_referenced: bool = False
+    fuzz_target_source: str = ""
+    build_script_source: str = ""
+
+    @property
+    def success(self) -> bool:
+        """Whether the build was successful (compiles and produces usable binary)."""
+        return self.compiles and self.binary_exists and self.is_function_referenced
+
+    def to_dict(self) -> Dict:
+        """Convert to a dictionary for serialization."""
+        return {
+            'compiles': self.compiles,
+            'build_success': self.success,
+            'log_path': self.log_path,
+            'errors': self.errors,
+            'binary_exists': self.binary_exists,
+            'is_function_referenced': self.is_function_referenced,
+            'fuzz_target_source': self.fuzz_target_source,
+            'build_script_source': self.build_script_source,
+        }
+
+
+@dataclass
+class RunInfo:
+    """Information about a fuzz target run."""
+    crashes: bool = False
+    log_path: str = ""
+    corpus_path: str = ""
+    reproducer_path: str = ""
+
+    def to_dict(self) -> Dict:
+        """Convert to a dictionary for serialization."""
+        return {
+            'crashes': self.crashes,
+            'log_path': self.log_path,
+            'corpus_path': self.corpus_path,
+            'reproducer_path': self.reproducer_path,
+        }
+
+
+@dataclass
+class CrashAnalysis:
```

Comment:
Do you plan to link the CrashAnalysis and CoverageAnalysis to the FuzzTargetResult class?
Or Would you plan to use FuzzTargetResult only for the current semantic analyzer and these two analyzers for analysis agents for now?

In the long-term, our plan to integrate the heuristics in semantic analyzer into crash analysis agent. For example, if a fuzz target crashed, we will first examine it with heuristics, then pass the fuzz target with the heuristic analysis result to crash analyzer agent.